src/lib/auxiliary_database/pagination.ml

```diff
@@ -6,84 +6,89 @@ let assert_same_set expected_transactions transactions =
   User_command.Set.(
     [%test_eq: t] (of_list expected_transactions) (of_list transactions))
 
-module Make (Value : sig
-  type t [@@deriving bin_io, compare, sexp, hash]
+module Make (Cursor : sig
```

Comment:
Typically the term `Cursor` usually means "some opaque pointer" that you can use to resume some sort of query. With this definition it seems like `Time.t` is the actual cursor here. This seems like something you're using to partition the data. Can you explain more about what this is? (maybe in person)

Reply:
Seems good since it's a pointer to something in a sorted structure